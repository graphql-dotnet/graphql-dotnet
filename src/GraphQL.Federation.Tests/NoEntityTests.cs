using GraphQL.Types;
using Microsoft.Extensions.DependencyInjection;

namespace GraphQL.Federation.Tests;

public class NoEntityTests
{
    [Fact]
    public void DoesNotCreateRepresentationsWhenNoResolvableTypes()
    {
        var services = new ServiceCollection();
        services.AddGraphQL(b => b
            .AddAutoSchema<MyQuery>()
            .AddFederation("2.3"));
        using var provider = services.BuildServiceProvider();
        var schema = provider.GetRequiredService<ISchema>();
        schema.Initialize();
        var sdl = schema.Print(new() { IncludeImportedDefinitions = false });
        sdl.ShouldMatchApproved(c => c.NoDiff());
    }

    private class MyQuery
    {
        public static Product FavoriteProduct => new Product { Id = 1 };
    }

    [Key("id", Resolvable = false)]
    private class Product
    {
        [Id]
        public int Id { get; set; }
    }
}
