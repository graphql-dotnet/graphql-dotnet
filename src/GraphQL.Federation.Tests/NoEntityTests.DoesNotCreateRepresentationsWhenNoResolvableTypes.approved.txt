schema @link(url: "https://specs.apollo.dev/link/v1.0", import: ["@link"]) @link(url: "https://specs.apollo.dev/federation/v2.3", import: ["@key", "@external", "@requires", "@provides", "@shareable", "@inaccessible", "@override", "@tag"]) {
  query: MyQuery
}

scalar _Any

type MyQuery {
  favoriteProduct: Product!
  _service: _Service!
}

type Product @key(fields: "id", resolvable: false) {
  id: ID!
}

type _Service {
  sdl: String
}